### PR TITLE
Implements integration with Sierra API to support Sierra credential validation

### PR DESCRIPTION
**Pull request by danielgrant**

### CODE REVIEW

File: packages/shared/sierra-client/src/index.ts [hunk 1]
```diff
@@ -77,57 +105,89 @@ export default class SierraClient {
...
 
-  private extractVarField(varFields: VarField[], fieldTag: string) {
+  private extractVarFieldContent(varFields: VarField[], fieldTag: string) {
     const found = varFields.find(varField => varField.fieldTag === fieldTag);
     return found ? found.content : '';
   }
 
-  private getPatronName(varField: string): PatronName {
-    if (!varField.trim()) {
+  private getPatronName(varFields: VarField[]) {
```
Comment (garyttierney) [inline packages/shared/sierra-client/src/index.ts:134]: I think this should go out into it's own module so we can unit test deserialization of Sierra records.

File: packages/shared/sierra-client/src/index.ts [hunk 2]
```diff
@@ -77,57 +105,84 @@ export default class SierraClient {
...
     }).then(response => {
-      return axios.create({
+      const instance = axios.create({
         baseURL: this.apiRoot,
         headers: {
           Authorization: 'Bearer ' + response.data.access_token
-        },
+        }
       });
+      instance.interceptors.response.use(response => response, error => this.unhandledError(error));
```
Comment (garyttierney) [inline packages/shared/sierra-client/src/index.ts:114]: Errors are turned into whatever `this.unhandledError(...)` is returning here, but the error is never caught by checking the promise for errors.

**Review by garyttierney — CHANGES_REQUESTED** [review]

File: packages/shared/sierra-client/src/index.ts (hunk 2 above)
Comment (danielgrant) [reply packages/shared/sierra-client/src/index.ts:114]: That's not quite right. This construct here is defining the default handler for anything which Axios construes as an erroneous response, i.e. request errors such as timeouts, or any response with an HTTP status code that doesn't match the `validateStatus` function. When such errors occur, the caller will receive our custom `ErrorResponse` type, with a `status` of `SierraStatus.UnknownError` and a descriptive error message in the `message` field.

There's no error / exceptions being thrown around here, so `try / catch` or `Promise`'s `.catch(...)` doesn't apply.

Comment (danielgrant) [reply packages/shared/sierra-client/src/index.ts:114]: With that being said, there is a problem here - `this.unhandledError(error)` will receive an `AxiosError`, not an `AxiosResponse`. Will fix that.

Comment (danielgrant) [reply packages/shared/sierra-client/src/index.ts]: I've updated the status code / error handling approach to resolve this.

File: packages/shared/sierra-client/src/index.ts (hunk 1 above)
Comment (danielgrant) [reply packages/shared/sierra-client/src/index.ts]: I've refactored the Patron models out into `patron.ts`.

**Review by garyttierney — APPROVED** [review]

**Pull request merged**